src/legacy/core_plugins/interpreter/public/functions/kibana.test.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { functionWrapper } from '../../test_helpers';
+import { kibana } from './kibana';
+
+describe('interpreter/functions#kibana', () => {
+  const fn = functionWrapper(kibana);
+  let context;
+  let initialContext;
+  let handlers;
+
+  beforeEach(() => {
+    context = { timeRange: { from: '0', to: '1' } };
+    initialContext = {
+      query: { language: 'lucene', query: 'geo.src:US' },
+      filters: [{ meta: { disabled: false }, query: { match: {} } }],
+      timeRange: { from: '2', to: '3' },
+    };
+    handlers = {
+      getInitialContext: () => initialContext,
+    };
+  });
+
+  it('returns an object with the correct structure', () => {
+    const actual = fn(context, {}, handlers);
+    expect(actual).toMatchSnapshot();
+  });
+
+  it('uses timeRange from context if not provided in initialContext', () => {
+    initialContext.timeRange = undefined;
+    const actual = fn(context, {}, handlers);
+    expect(actual.timeRange).toEqual({ from: '0', to: '1' });
+  });
+
+  it.skip('combines query from context with initialContext', () => {
+    context.query = { language: 'kuery', query: 'geo.dest:CN' };
+    // TODO: currently this fails & likely requires a fix in run_pipeline
```

Comment:
The issue here is this: https://github.com/elastic/kibana/blob/f1a02d19f93e9b54206baf195be67ffcd6a0f5cf/src/legacy/core_plugins/interpreter/public/functions/kibana.js#L33-L35

The `kibana` function expects you to be able to `concat` the query from `initialContext` with the one from `context.query`, but this fails as the one from `getInitialContext` is not an array.

Reply:
this is something that still needs to be defined .... array or not. I have no strong feeling about it